src/marinade/infos.ts

```diff
@@ -0,0 +1,165 @@
+import { Connection, PublicKey } from "@solana/web3.js";
+
+import { MARINADE_PROGRAM_ID, MSOL_MINT_ADDRESS, MSOL_VAULT_ADDRESS } from "./ids";
+
+import { IInstanceVault, IVaultInfoWrapper } from "../types";
+import { MARINADE_FINANCE_ACCOUNT_STATE } from "./layouts";
+
+import * as types from ".";
+import { ASSOCIATED_TOKEN_PROGRAM_ID, TOKEN_PROGRAM_ID } from "@solana/spl-token-v2";
+
+let infos: IInstanceVault;
+
+infos = class InstanceMarinade {
+  static async getAllVaults(connection: Connection): Promise<types.VaultInfo[]> {
+    const accountInfos = await connection.getAccountInfo(MSOL_VAULT_ADDRESS);
+
+    if (!accountInfos) throw Error("Error: Could not get solido address");
+    const vault = this.parseVault(accountInfos.data, MSOL_VAULT_ADDRESS);
+
+    return [vault];
+  }
+
+  static async getAllVaultWrappers(connection: Connection): Promise<VaultInfoWrapper[]> {
+    return (await this.getAllVaults(connection)).map((vault) => new VaultInfoWrapper(vault));
+  }
+
+  static async getVault(connection: Connection, vaultId: PublicKey): Promise<types.VaultInfo> {
+    if (!vaultId.equals(MSOL_VAULT_ADDRESS)) {
+      throw Error(`Error: Marinade vaultId must match ${MSOL_VAULT_ADDRESS.toBase58()}`);
+    }
+
+    const vaultAccountInfo = await connection.getAccountInfo(vaultId);
+    if (!vaultAccountInfo) throw Error("Error: Cannot get marinade token account");
+
+    const vault: types.VaultInfo = this.parseVault(vaultAccountInfo.data, vaultId);
+
+    return vault;
+  }
+
+  static parseVault(data: Buffer, vaultId: PublicKey): types.VaultInfo {
+    // Decode the Token data using AccountLayout
+
+    // const testLayout = struct(
+    //   [
+    //     blob(8, "identifier"),
+    //     publicKey("msolMint"),
+    //     publicKey("adminAuthority"),
+    //     publicKey("operationalSolAccount"),
+    //     publicKey("treasuryMsolAccount"),
+    //     u8("reserveBumpSeed"),
+    //   ],
+    //   "State"
+    // );
+
+    const decodeData = MARINADE_FINANCE_ACCOUNT_STATE.decode(data);
+    console.log({ decodeData });
```

Comment:
Plz check all data type is correct 
Some properties type are `number` instead of `BN`.

Reply:
Also plz remove debug message after done fixing